src/ngraph/frontend/onnx_import/op/lstm.cpp

```diff
@@ -0,0 +1,369 @@
+                {
+                    LSTM_INPUT_X,
+                    LSTM_INPUT_W,
+                    LSTM_INPUT_R,
+                    LSTM_INPUT_B,
+                    LSTM_INPUT_SEQ_LENGTHS,
+                    LSTM_INPUT_INIT_H,
+                    LSTM_INPUT_INIT_C,
+                    LSTM_INPUT_P
+                };
+
+                std::string to_str(const LSTMInput& in)
+                {
+                    switch (in)
+                    {
+                    case LSTMInput::LSTM_INPUT_X: return "X";
+                    case LSTMInput::LSTM_INPUT_W: return "W";
+                    case LSTMInput::LSTM_INPUT_R: return "R";
+                    case LSTMInput::LSTM_INPUT_B: return "B";
+                    case LSTMInput::LSTM_INPUT_SEQ_LENGTHS: return "sequence_lens";
+                    case LSTMInput::LSTM_INPUT_INIT_H: return "initial_h";
+                    case LSTMInput::LSTM_INPUT_INIT_C: return "initial_c";
+                    case LSTMInput::LSTM_INPUT_P: return "P";
+                    default: return "Unrecognized input value!";
+                    }
+                }
+
+                struct LSTMNgInputMap
+                {
+                    using container_type = std::map<LSTMInput, std::shared_ptr<ngraph::Node>>;
+                    using iterator = typename container_type::iterator;
+
+                    explicit LSTMNgInputMap(const Node& node)
+                    {
+                        const auto& ng_inputs = node.get_ng_inputs();
+                        // We have input, output, forget and cell gates
+                        constexpr std::size_t gates_count{4};
+                        // Peepholes add additional connections to input, output and forget gates.
+                        constexpr std::size_t peepholes_count{3};
+
+                        // ----- Mandatory inputs ------
+                        // Packed input sequences. Shape: [seq_length, batch_size, input_size]
+                        m_map[LSTMInput::LSTM_INPUT_X] = ng_inputs.at(0);
+                        // Weight tensor for the gates. Shape: [num_directions, 4*hidden_size, input_size]
+                        m_map[LSTMInput::LSTM_INPUT_W] = ng_inputs.at(1);
+                        // The recurrence weight tensor. Shape: [num_directions, 4*hidden_size, hidden_size]
+                        m_map[LSTMInput::LSTM_INPUT_R] = ng_inputs.at(2);
+
+                        const std::size_t hidden_size =
+                            m_map[LSTMInput::LSTM_INPUT_R]->get_shape().back();
+                        const std::size_t batch_size =
+                            m_map[LSTMInput::LSTM_INPUT_X]->get_shape().at(1);
+                        const std::size_t num_directions =
+                            m_map[LSTMInput::LSTM_INPUT_W]->get_shape().front();
+
+                        // ------ Optional inputs ------
+                        // The bias tensor for input gate. Shape [num_directions, 8*hidden_size]
```

Comment:
isn't the bias shape should be ` (4*hidden_size)`

Reply:
You're partially right, however this is a:
>  Concatenation of `[Wb[iofc], Rb[iofc]]`
as stated here: https://github.com/onnx/onnx/blob/master/docs/Operators.md#inputs-3---8